### PR TITLE
mkcomposefs: Add missing free()

### PR DESCRIPTION
**Pull request by eriksjolund**

### CODE REVIEW

File: tools/mkcomposefs.c [hunk 1]
```diff
@@ -129,6 +129,7 @@ static char *unescape_string(const char *escaped, size_t escaped_size,
 		if (c == '\\') {
 			if (escaped >= escaped_end) {
 				*err = make_error("No character after escape");
+				free(res);
```
Comment (cgwalters) [inline tools/mkcomposefs.c:132]: Let's use `cleanup_free` instead, it's designed for this.  The return path will need to `steal_pointer`.

**Review by cgwalters — CHANGES_REQUESTED** [review]
This is OK but I really prefer using the cleanup attributes as a general rule because it makes error handling *much* more reliable.

**Comment by eriksjolund** [comment]
> This is OK but I really prefer using the cleanup attributes as a general rule because it makes error handling _much_ more reliable.

Ok, I'll give it a try.

**Review by cgwalters — APPROVED** [review]
Thanks!  I hope you'd agree this is a less error-prone as code is refactored.

**Comment by eriksjolund** [comment]
Yes, this is better.

**Pull request merged**